src/main/java/com/exasol/containers/ExasolDockerImageReference.java

```diff
@@ -15,12 +15,38 @@
             Pattern.compile("(?:(?:exasol/)?(?:docker-db:))?" // prefix (optional)
                     + "(\\d+)(?:\\.(\\d+))?(?:\\.(\\d+))?" // Exasol version (partially optional)
                     + "(?:-d(\\d+))?"); // docker image revision (optional)
-    private static final Pattern MAJOR_VERSION_PATTERN = Pattern
-            .compile(Pattern.quote(EXASOL_DOCKER_IMAGE_ID) + ":(\\d+).*");
+    private static final int VERSION_NOT_PRESENT = -1;
     private final String reference;
+    private final int major;
+    private final int minor;
+    private final int fix;
+    private final int dockerImageRevision;
 
     private ExasolDockerImageReference(final String reference) {
         this.reference = reference;
+        this.major = VERSION_NOT_PRESENT;
+        this.minor = VERSION_NOT_PRESENT;
+        this.fix = VERSION_NOT_PRESENT;
+        this.dockerImageRevision = VERSION_NOT_PRESENT;
+    }
+
+    private ExasolDockerImageReference(final int major, final int minor, final int fix, final int dockerImageRevision) {
```

Comment:
I think now this class is hard to understand because it does too many things at the same time.
Even more, it has redundancy to store reference and its components in parallel. 
My suggestion: Create interface: ExasolDockerImageReference and two implementors OffialExasolDockerImageReference and OtherDockerImageReference.
The official parses and splits the values, the other only stores a string.
In addition, we could add a factory that builds the appropriate class depending on the input string.

Reply:
Done.